cmd/gonic/gonic.go

```diff
@@ -38,7 +39,12 @@ func main() {
 	confTLSKey := set.String("tls-key", "", "path to TLS private key (optional)")
 	confPodcastPath := set.String("podcast-path", "", "path to podcasts")
 	confCachePath := set.String("cache-path", "", "path to cache")
-	confDBPath := set.String("db-path", "gonic.db", "path to database (optional)")
+	confSqlitePath := set.String("db-path", "gonic.db", "path to database (optional, default: gonic.db)")
+	confPostgresHost := set.String("postgres-host", "", "name of the PostgreSQL gonicServer (optional)")
+	confPostgresPort := set.Int("postgres-port", 5432, "port to use for PostgreSQL connection (optional, default: 5432)")
+	confPostgresName := set.String("postgres-db", "gonic", "name of the PostgreSQL database (optional, default: gonic)")
+	confPostgresUser := set.String("postgres-user", "gonic", "name of the PostgreSQL user (optional, default: gonic)")
+	confPostgresSslModel := set.String("postgres-ssl-mode", "verify-full", "the ssl mode used for connecting to the PostreSQL instance (optional, default: verify-full)")
```

Comment:
i think i would prefer we just have two new options like

```
-db-driver
-db-dsn
```

then we can pass those directly to gorm.Open()

what do you think? it would give us all the flexibility and future compatibility 

for that to work we just need
- a bit of logic in main to check that "db-path" is present, print a deprecation message, and just set driver to sqlite and dsn to the file name
- db.New() to have a special case for sqlite to add the DefaultOptions() and SetMaxOpenConns to the dsn url (the file:/// thing) 
- any thing else?

Reply:
That would mean our users would need to understand the GORM DSNs, which is not ideal but also not the end of the world. The bigger concern I have is that the SQL used is slightly different across the DBs (as there are already adjustments between SQLite and PostgreSQL), so saying all will work seems bold.

If you feel that is ok and the testing limits are more a documentation thing, that works for me though.